Tiltfile

```diff
@@ -396,6 +399,120 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create flavor resources from cluster-template files in the templates directory
+def flavors():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    # TODO: should we set `./test/infrastructure/docker/templates`as the default?
+    flavor_dirs = settings.get("flavor_dirs", [])
+
+    for flavor_dir in flavor_dirs:
+        template_list = [item for item in listdir(flavor_dir)]
+        template_list = [template_name for template_name in template_list if os.path.basename(template_name).endswith("yaml")]
```

Comment:
Can we filter on:
* starts with `cluster-template`
* ends with `.yaml`

instead?

I think this is necessary for folders which also contain ClusterClasses.

Reply:
I'm experimenting with sorting the flavors into Clusters and ClusterClasses like this so I don't think we need to filter out ClusterClass templates based on the filename. Wdyt?

![image](https://user-images.githubusercontent.com/17480384/157326980-7819cdb1-9685-4b3f-90fb-ad6e566cc9bf.png)

Reply:
I think there is a misunderstanding.

The only cluster-templates/flavors we have are `cluster-template.yaml` and `cluster-template-${flavor}.yaml`.

ClusterClasses itself are not flavors. `clusterctl generate cluster` renders ClusterClasses automatically when they are used in one of the cluster-templates/flavors

Reply:
Yup +1 to what @sbueringer said, and also we shouldn't explicitly separate "cluster class flavors" in the UI. ClusterClass is just a feature that is used in some cluster template flavors and shouldn't be treated differently from other flavors that don't use ClusterClasses.